packages/firestore/src/local/indexeddb_remote_document_cache.ts

```diff
@@ -178,6 +183,72 @@ export class IndexedDbRemoteDocumentCache implements RemoteDocumentCache {
       });
   }
 
+  getEntries(
+    transaction: PersistenceTransaction,
+    documentKeys: DocumentKeySet
+  ): PersistencePromise<NullableMaybeDocumentMap> {
+    return this.getSizedEntries(transaction, documentKeys).next(
+      result => result.maybeDocuments
+    );
+  }
+
+  /**
+   * Looks up several entries in the cache.
+   *
+   * @param documentKeys The set of keys entries to look up.
+   * @return A map of MaybeDocuments indexed by key (if a document cannot be
+   *     found, the key will be mapped to null) and a map of sizes indexed by
+   *     key (zero if the key cannot be found).
+   */
+  getSizedEntries(
+    transaction: PersistenceTransaction,
+    documentKeys: DocumentKeySet
+  ): PersistencePromise<DocumentSizeEntries> {
+    let results = nullableMaybeDocumentMap();
+    let sizeMap = new SortedMap<DocumentKey, number>(DocumentKey.comparator);
+    if (documentKeys.isEmpty()) {
+      return PersistencePromise.resolve({ maybeDocuments: results, sizeMap });
+    }
+
+    const range = IDBKeyRange.bound(
+      documentKeys.first()!.path.toArray(),
+      documentKeys.last()!.path.toArray()
+    );
+    let key = documentKeys.first();
+
+    return remoteDocumentsStore(transaction)
+      .iterate({ range }, (potentialKeyRaw, dbRemoteDoc, control) => {
+        const potentialKey = DocumentKey.fromSegments(potentialKeyRaw);
+        while (DocumentKey.comparator(key!, potentialKey) != 1) {
+          if (key!.isEqual(potentialKey)) {
+            results = results.insert(
+              key!,
+              this.serializer.fromDbRemoteDocument(dbRemoteDoc)
+            );
+            sizeMap = sizeMap.insert(key!, dbDocumentSize(dbRemoteDoc));
```

Comment:
Whether to update the `sizeMap` could be controlled by a flag to avoid doing unnecessary work when this function is invoked via `getEntries`, but it might be overkill.